File: backend/src/main/java/com/interviewace/backend/service/JobMatchService.java
package com.interviewace.backend.service;
import com.interviewace.backend.dto.JobMatchAnalysisDto;
import com.interviewace.backend.dto.JobMatchRequest;
import java.util.List;
public interface JobMatchService {
    JobMatchAnalysisDto analyzeJobMatch(Long resumeId, JobMatchRequest request);
    List<JobMatchAnalysisDto> getJobMatchHistory(Long resumeId);
    JobMatchAnalysisDto getLatestJobMatch(Long resumeId);
}

File: frontend/src/services/jobMatchService.ts
import { api } from './api'
import type { JobMatchAnalysis, JobMatchRequest } from '../types/jobMatch'

export const jobMatchService = {
  async analyzeJobMatch(resumeId: number, jobDescription: string) {
    const body: JobMatchRequest = { jobDescription }
    return (await api.post<JobMatchAnalysis>(`/resumes/${resumeId}/job-match`, body)).data
  },
  async getLatestJobMatch(resumeId: number) {
    return (await api.get<JobMatchAnalysis>(`/resumes/${resumeId}/job-match/latest`)).data
  },
  async getJobMatchHistory(resumeId: number) {
    return (await api.get<JobMatchAnalysis[]>(`/resumes/${resumeId}/job-matches`)).data
  },
}
